src/Iris/Colour/Detect.hs

```diff
@@ -0,0 +1,28 @@
+{- |
+Module                  : Iris.Colour.Detect
+Copyright               : (c) 2020 Kowainik
+                          (c) 2022 Dmitrii Kovanikov
+SPDX-License-Identifier : MPL-2.0
+Maintainer              : Dmitrii Kovanikov <kovanikov@gmail.com>
+Stability               : Experimental
+Portability             : Portable
+
+Detects if colour should be enabled or not
+
+@since x.x.x.x
+-}
+
+module Iris.Colour.Detect
+    ( detectColour
+    ) where
+import System.Environment (lookupEnv)
+
+
+
+detectColour :: Maybe String -> IO Bool
+detectColour appname = any  (/= Nothing) <$> traverse lookupEnv varnames
```

Comment:
Currently, this function misses the check of the `TERM=dumb` environment variable